src/ddmd/backend/cod3.c

```diff
@@ -945,8 +945,18 @@ void outblkexitcode(CodeBuilder& cdb, block *bl, int& anyspill, const char* sfls
         }
 
         case BC_try:
+#if MARS
+            if (config.ehmethod == EH_NONE)
+            {
+            }
+            else
+            if (config.ehmethod == EH_WIN32)
+#endif
+#if SCPP
             if (config.exe == EX_WIN32)
-            {   usednteh |= NTEH_try;
+#endif
```

Comment:
Oy this use of the preprocessor is quite nasty... far as I understand exactly one of MARS or SCPP must be defined. Then this is quite a bit easier on the eyes:

```C++
#if MARS
            if (config.ehmethod == EH_NONE)
            {
            }
            else
            if (config.ehmethod == EH_WIN32)
#else
            if (config.exe == EX_WIN32)
#endif
```

Reply:
I avoid that style now because MARS isn't on-off. The back end supports multiple separate builds - SCPP, SPP, HTOD, and MARS.

Reply:
All I'm saying is the code doesn't work unless exactly one of MARS and SCPP is defined, and that's difficult to infer from the code.

Reply:
I know, but in the rest of the code I removed the `#else` because of bugs and confusion about what code was for which build.

Reply:
This particular piece of code is ugly, and I'm going to straighten it out in a follow-on, as it will require some refactoring. In the meantime, it gets the RAII working.